src/types.ts

```diff
@@ -107,6 +107,35 @@ export type AppSettingsAPIResponse<
   };
 };
 
+export type MessageFlagsResponse<
+  ChannelType extends UnknownType = UnknownType,
+  CommandType extends string = LiteralStringForUnion,
+  UserType extends UnknownType = UnknownType,
+  AttachmentType = UnknownType,
+  MessageType = UnknownType,
+  ReactionType = UnknownType
+> = APIResponse & {
+  flags?: Array<{
+    message: MessageResponse<
+      AttachmentType,
+      ChannelType,
+      CommandType,
+      MessageType,
+      ReactionType,
+      UserType
+    >;
+    user: UserResponse<UserType>;
+    approved_at?: string;
+    created_at?: string;
+    created_by_automod?: boolean;
+    moderation_result?: {};
```

Comment:
this is not 100% accurate, could be `UnknownType` if we don't know the exact type

Reply:
created a new PR for this here: https://github.com/GetStream/stream-chat-js/pull/681